src/Process/Listener/Command.php

```diff
@@ -18,18 +18,21 @@ public function hasMessages(): bool
         return $this->_getMessageBroker()->hasMessage();
     }
 
-    public function processMessages(): ListenerInterface
+    public function processMessage(): ListenerInterface
     {
-        $message = $this->_getMessageBroker()->getNextMessage();
-        if (json_decode($message) !== null) {
-            $this->_getExpressionLanguage()->evaluate(
-                json_decode($message, true)['command'],
-                [
-                    'commandProcess' => $this,
-                ]
-            );
-        } else {
-            $this->_getLogger()->warning('The message is not a JSON: "' . $message . '".');
+        $message = $this->_getMessageBroker()->attemptGetNextMessage();
```

Comment:
@mucha55 - thinking about it more, truthfully the message could be lost if the process crashes here or in `attemptGetNextMessage`. 

So, it's really more of a matter of a smaller unprotected critical region as opposed to memoizing the message in `hasMessage`. The lossless solution would be far more complicated, making the message "not visible" in Redis and then forcing it to correctly be deleted or become "visible" again IFF the process dies which could be identified using the Process's UUID. Again that is almost certainly a future us thing.

However, with that knowledge, I think that I would prefer not using `?string` and instead memoizing the message in `hasMessage` and then consuming it with `getNextMessage` - additionally making that method logically defensive.

Of course this is all totally up to you two, but I like that it removes `?string` for the cost of a slightly larger (and unprotected in both cases) critical region.

Also, I am aware that initially I steered you in this direction and am now changing my mind b/c I'm a space cadet.

Reply:
I'll open another PR with the implementation I mentioned in the description so you guys can compare them side by side